### PR TITLE
Remove redundant getRemoteAddress added by Paper that causes an error

### PR DESCRIPTION
**Pull request by PureGero**
The original code for getting a player's ip address is as follows:
```java
String s1 = "local";

if (connection.getRemoteAddress() != null) {
    s1 = connection.getRemoteAddress().toString();
}
```

However, this s1 variable became unused and the following was used instead:
```java
postChunkLoadJoin( ... connection.getRemoteAddress().toString() ... )
```

This new code does not check if the `connection.getRemoteAddress()` is null, and throws a NullPointerException when it is null.

I have fixed this by replacing the new code with the original s1 variable. However, postChunkLoadJoin is run inside a lambda function, thus an effectively final copy of s1 was needed to be created.
```java
String finalS1 = s1;
postChunkLoadJoin( ... finalS1 ... )
```

### CODE REVIEW

**Comment by MiniDigger** [comment]
can't you do smth like
```
final String s1;

if (connection.getRemoteAddress() != null) {
    s1 = connection.getRemoteAddress().toString();
} else {
   s1 = "local";
}
```
instead?

File: patches/server/0819-Add-config-option-for-logging-player-ip-addresses.patch [hunk 1]
```diff
@@ -93,12 +93,3 @@ index 99aa8f2ba2f10578f37de621d1a5a8e222cd70b1..eaa005c1c9b4386bcdbe1d6eb28c3eca
          }
  
          // Spigot start - spawn location event
```
Comment (Proximyst) [inline patches/server/0819-Add-config-option-for-logging-player-ip-addresses.patch]: Am I being blind or did you just entirely break another patch?

**Review by Proximyst — DISMISSED** [review]

**Comment by PureGero** [comment]
> can't you do smth like
> 
> ```
> final String s1;
> 
> if (connection.getRemoteAddress() != null) {
>     s1 = connection.getRemoteAddress().toString();
> } else {
>    s1 = "local";
> }
> ```
> 
> instead?

That is possible, but that'd mean modifying upstream code which isn't recommended right? Minimal modifications are preferred

File: patches/server/0819-Add-config-option-for-logging-player-ip-addresses.patch (hunk 1 above)
Comment (PureGero) [reply patches/server/0819-Add-config-option-for-logging-player-ip-addresses.patch]: s1 uses the exact same code, that patch duplicated the same code twice

Comment (PureGero) [reply patches/server/0819-Add-config-option-for-logging-player-ip-addresses.patch]: That patch also includes:
```java
         if (connection.getRemoteAddress() != null) {
-            s1 = connection.getRemoteAddress().toString();
+            s1 = com.destroystokyo.paper.PaperConfig.logPlayerIpAddresses ? connection.getRemoteAddress().toString() : "<ip address withheld>"; // Paper
         }
```

**Comment by Proximyst** [comment]
> Aren't minimal
> 
> > can't you do smth like
> > ```
> > final String s1;
> > 
> > if (connection.getRemoteAddress() != null) {
> >     s1 = connection.getRemoteAddress().toString();
> > } else {
> >    s1 = "local";
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > instead?
> 
> That is possible, but that'd mean modifying upstream code which isn't recommended right? Minimal modifications are preferred

It’d place a diff on change on it, which is also a good idea in places such as this where the variable name is not descriptive and probably just generated. I’d say change upstream :)

**Comment by PureGero** [comment]
> > > can't you do smth like
> > > ```
> > > final String s1;
> > > 
> > > if (connection.getRemoteAddress() != null) {
> > >     s1 = connection.getRemoteAddress().toString();
> > > } else {
> > >    s1 = "local";
> > > }
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > instead?
> > 
> > 
> > That is possible, but that'd mean modifying upstream code which isn't recommended right? Minimal modifications are preferred
> 
> It’d place a diff on change on it, which is also a good idea in places such as this where the variable name is not descriptive and probably just generated. I’d say change upstream :)

I've changed the upstream, should I also rename s1 to something more descriptive? That'll involve modifying a patch before it and a line of code that hasn't had a diff on it yet if I rename all references in the file.

**Comment by Proximyst** [comment]
> > > > can't you do smth like
> > > > ```
> > > > final String s1;
> > > > 
> > > > if (connection.getRemoteAddress() != null) {
> > > >     s1 = connection.getRemoteAddress().toString();
> > > > } else {
> > > >    s1 = "local";
> > > > }
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > instead?
> > > 
> > > 
> > > That is possible, but that'd mean modifying upstream code which isn't recommended right? Minimal modifications are preferred
> > 
> > 
> > It’d place a diff on change on it, which is also a good idea in places such as this where the variable name is not descriptive and probably just generated. I’d say change upstream :)
> 
> I've changed the upstream, should I also rename s1 to something more descriptive? That'll involve modifying a patch before it and a line of code that hasn't had a diff on it yet if I rename all references in the file.

No, changing the name isn’t necessary. :)

File: patches/server/0416-Load-Chunks-for-Login-Asynchronously.patch [hunk 2]
```diff
@@ -136,7 +136,24 @@ index 9609579e6d61c3975740d07a394680aeef952516..6b23b9389ff92ae8016d4adb289ac2a0
...
  
          if (nbttagcompound != null) {
-@@ -260,6 +267,52 @@ public abstract class PlayerList {
+@@ -213,11 +220,15 @@ public abstract class PlayerList {
+         if (nbttagcompound == null) player.fudgeSpawnLocation(worldserver1); // Paper - only move to spawn on first login, otherwise, stay where you are....
+ 
+         player.setLevel(worldserver1);
+-        String s1 = "local";
++        // Paper start - make s1 effectively final
++        String s1;
```
Comment (Proximyst) [inline patches/server/0416-Load-Chunks-for-Login-Asynchronously.patch]: Explicitly make it final.

**Review by Proximyst — CHANGES_REQUESTED** [review]

File: patches/server/0416-Load-Chunks-for-Login-Asynchronously.patch (hunk 2 above)
Comment (PureGero) [reply patches/server/0416-Load-Chunks-for-Login-Asynchronously.patch]: Done, thank you

**Review by Machine-Maker — APPROVED** [review]
Needs a rebase, but changes look good
